### PR TITLE
feat(Entity): add removeMany by predicate

### PR DESCRIPTION
**Pull request by timdeschryver**
This PR allows to remove entities based on a predicate.
See #672

### CODE REVIEW

**Comment by MikeRyanDev** [comment]
This looks great to me!

I am going to label this at v6.x. We are in the middle of adding new infrastructure to support multiple versions in the same repo. Please don't be surprised if this doesn't get merged immediately.

**Comment by tdeschryver** [comment]
@MikeRyanDev Should this change (and the update many from #907) also be documented in the current docs? 
If so, I'll edit the docs with an extra `Action`.

**Comment by brandonroberts** [comment]
@timdeschryver will you rebase this so it can land?

**Comment by timdeschryver** [comment]
✔️

File: modules/entity/src/models.ts [hunk 1]
```diff
@@ -38,6 +38,10 @@ export type UpdateNum<T> = {
 
 export type Update<T> = UpdateStr<T> | UpdateNum<T>;
 
+export type Predicate<T> = {
```
Comment (brandonroberts) [inline modules/entity/src/models.ts]: Make this the same as `export type ComparerStr<T> = (a: T, b: T) => string;`

**Pull request merged**